### PR TITLE
proper implementation for forms (FormUrlEncodedContent)

### PR DESCRIPTION
**Pull request by artur-siwiak**
If base request contains Content Type = 'application/x-www-form-urlencoded' then value of those are not copied to then new request

### CODE REVIEW

File: src/Core/Helpers.cs [hunk 1]
```diff
@@ -60,21 +60,37 @@ private static HttpRequestMessage CreateProxyHttpRequest(this HttpContext contex
...
+            if (requestMessage.Content != null)
+            {
+                foreach (var header in request.Headers)
+                {
+                    if (!requestMessage.Headers.TryAddWithoutValidation(header.Key, header.Value.ToArray()))
+                    {
+                        requestMessage.Content?.Headers.TryAddWithoutValidation(header.Key, header.Value.ToArray());
+                    }
+                }
+            }
```
Comment (twitchax) [inline src/Core/Helpers.cs]: Why did you move this?  Also, please remove braces: I know some people like that standard, but I do not.

File: src/Core/Helpers.cs [hunk 2]
```diff
@@ -60,21 +60,37 @@ private static HttpRequestMessage CreateProxyHttpRequest(this HttpContext contex
...
                 !HttpMethods.IsDelete(requestMethod) &&
                 !HttpMethods.IsTrace(requestMethod))
             {
-                var streamContent = new StreamContent(request.Body);
-                requestMessage.Content = streamContent;
+                if (request.HasFormContentType)
+                {
+                    requestMessage.Content = new FormUrlEncodedContent(
+                        request.Form
+                            .Select(x => new KeyValuePair<string, StringValues>(x.Key, x.Value)).ToDictionary(x => x.Key, x => x.Value.ToString()));
```
Comment (twitchax) [inline src/Core/Helpers.cs]: Can you make this a little more readable?  Maybe...

```csharp
requestMessage.Content = new FormUrlEncodedContent(
    request.Form
        .Select(x => new KeyValuePair<string, StringValues>(x.Key, x.Value))
        .ToDictionary(x => x.Key, x => x.Value.ToString())
);
```

Comment (twitchax) [reply src/Core/Helpers.cs]: Also, can't you skip the `Select` clause?  I think you could just call `ToDictionary`, though I'd have to look at my intellisense probably.

File: src/Test/UnitTests.cs [hunk 3]
```diff
@@ -55,6 +56,21 @@ public async Task ProxyAttributePostRequest()
             Assert.Contains("101", JObject.Parse(responseString).Value<string>("id"));
         }
 
+        [Fact]
+        public async Task ProxyAttributePostWithFormRequest()
```
Comment (twitchax) [inline src/Test/UnitTests.cs]: Thanks for adding a test! :)

**Comment by twitchax** [comment]
@artur-siwiak, thanks for the PR!  I had a few comments if you don't mind taking a look at them.

File: src/Core/Helpers.cs (hunk 2 above)
Comment (artur-siwiak) [reply src/Core/Helpers.cs]: Yep, you have a right

File: src/Core/Helpers.cs (hunk 1 above)
Comment (artur-siwiak) [reply src/Core/Helpers.cs]: For some reason (sorry, I couldn't find why) after setting headers, FormUrlEncodedContent was kinda omitted and the code simply didn't work.

File: src/Test/UnitTests.cs (hunk 3 above)
Comment (artur-siwiak) [reply src/Test/UnitTests.cs]: Tests are included in project so they should be updated also ;)

File: src/Core/Helpers.cs (hunk 1 above)
Comment (twitchax) [reply src/Core/Helpers.cs]: Ahhh, interesting.  I bet something is overwritten.  Sounds good.
